processor/resourcedetectionprocessor/internal/system/system.go

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package system
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+)
+
+const (
+	// TypeStr is the detector type string
+	TypeStr = "system"
+)
+
+var _ internal.Detector = (*Detector)(nil)
+
+// Detector is a system metadata detector
+type Detector struct {
+	provider systemMetadata
+}
+
+// NewDetector creates a new system metadata detector
+func NewDetector() (internal.Detector, error) {
+	return &Detector{provider: &systemMetadataImpl{}}, nil
+}
+
+// Detect detects system metadata and returns a resource with the available ones
+func (d *Detector) Detect(ctx context.Context) (pdata.Resource, error) {
+	res := pdata.NewResource()
+	res.InitEmpty()
+	attrs := res.Attributes()
+
+	hostname, err := d.provider.Hostname()
+	if err != nil {
+		return res, err
+	}
+
+	hostType, err := d.provider.HostType()
+	if err != nil {
+		return res, err
+	}
+
+	if d.provider.FQDNAvailable() {
+		fqdn, err := d.provider.FQDN(ctx)
+		if err != nil {
+			return res, err
+		}
+
+		attrs.InsertString(conventions.AttributeHostName, fqdn)
+	}
+
+	attrs.InsertString(conventions.AttributeHostHostname, hostname)
+	attrs.InsertString(conventions.AttributeHostType, hostType)
```

Comment:
So I re-read the spec (turns out I was reading and old version thanks to Github search) and I think it makes more sense to remove this and instead have `os.type` mapped to `runtime.GOOS`. 

If there is interest at some point we could pursue a `os.architecture` resource convention that can be added here, but for now I think it's preferable to have `os.type`.